pkg/devconsole/webhooks/webhooks.go

```diff
@@ -0,0 +1,216 @@
+package webhooks
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/openshift/console/pkg/auth"
+	"github.com/openshift/console/pkg/devconsole/common"
+)
+
+func CreateGithubWebhook(r *http.Request, user *auth.User) (common.DevConsoleCommonResponse, error) {
+	// POST /api/dev-console/webhooks/github
+	var request GithubWebhookRequest
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to parse request: %v", err)
+	}
+
+	bodyBytes, err := json.Marshal(request.Body)
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to marshal request body: %v", err)
+	}
+
+	GH_WEBHOOK_URL := fmt.Sprintf("%s/repos/%s/%s/hooks", request.HostName, request.Owner, request.RepoName)
+
+	var serviceRequest *http.Request
+	serviceRequest, err = http.NewRequest(http.MethodPost, GH_WEBHOOK_URL, bytes.NewReader(bodyBytes))
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to create request: %v", err)
+	}
+
+	for key, values := range request.Headers {
+		for _, value := range values {
+			serviceRequest.Header.Add(key, value)
+		}
+	}
+
+	if request.AllowAuthHeader {
+		serviceRequest.Header.Set("Authorization", fmt.Sprintf("Bearer %s", user.Token))
+	}
```

Comment:
Same question here on passing the user's token here. Presumably these requests are going to Github? Why would we ever pass the user's token through?